parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -70,13 +70,13 @@ pub enum Destination {
 	ForeignAccountId20 { para_id: u32, id: [u8; 20] },
 }
 
-impl From<MessageV1> for Xcm<()> {
-	fn from(val: MessageV1) -> Self {
+impl<G: Get<[u8; 2]>> From<MessageV1<G>> for Xcm<()> {
+	fn from(val: MessageV1<G>) -> Self {
```

Comment:
See man comment above, I think we should introduce a separate converter type, ie `Converter::convert(msgv1) -> Xcm`